### PR TITLE
Add addons for dns,dashboard,gpu

### PR DESCRIPTION
**Pull request by evilnick**

### CODE REVIEW

File: docs/addon-dashboard.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+have an IP address on your local network, the recommended way to do this is
+through the proxy service. You can initiate the proxy with the command:
+
+```bash
+microk8s.kubectl proxy --accept-hosts=.* --address=0.0.0.0 &
+```
+
+You can then access the Dashboard at the address
+
+[http://127.0.0.1:8001/api/v1/namespaces/kube-system/services/https:kubernetes-dashboard:/proxy/]()
```
Comment (evilnick) [inline docs/addon-dashboard.md]: @ktsakalozos This URL no longer seems to work - can you confirm what URL should be used for proxy access?

File: docs/addon-dashboard.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+![IMAGE of Dashboard](#ref)
```
Comment (evilnick) [inline docs/addon-dashboard.md]: @evilnick image to be added

File: docs/addon-dashboard.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+---
+
+# Dashboard addon
+
+The standard Kubernetes Dashboard is a convenient way to keep track of the
+activity and resource use of MicroK8s
+
+To log in to the Dashboard, you will need the access token (unless RBAC has
+also been enabled). This is generated randomly on deployment, so a few commands
+are needed to retrieve it:
```
Comment (ktsakalozos) [inline docs/addon-dashboard.md]: We have to retreive the token after the addon gets enabled. Contrary to the docs for other addons (eg fluentd) here we do not say that the dashboard is enabled with `microk8s.enable dashboard`.

File: docs/addon-dashboard.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+```bash
+token=$(microk8s.kubectl -n kube-system get secret | grep default-token | cut -d " " -f1)
+microk8s.kubectl -n kube-system describe secret $token
+```
+In an RBAC enabled setup (microk8s.enable RBAC) you need to create a user with
```
Comment (ktsakalozos) [inline docs/addon-dashboard.md]: Formatting around `microk8s.enable rbac`? Note that RBAC should be lowercase.

File: docs/addon-dashboard.md (hunk 1 above)
Comment (ktsakalozos) [reply docs/addon-dashboard.md]: This works for me. The `microk8s.kubectl proxy` creates a proxy by default on 8001. What is the error you are getting?

Comment (evilnick) [reply docs/addon-dashboard.md]: I get this at that url:

{
  "kind": "Status",
  "apiVersion": "v1",
  "metadata": {
    
  },
  "status": "Failure",
  "message": "no endpoints available for service \"kubernetes-dashboard\"",
  "reason": "ServiceUnavailable",
  "code": 503
}

so the proxy is working, and the apiserver is there, but the endpoint doesn't seem to be

File: docs/addon-dashboard.md (hunk 3 above)
Comment (evilnick) [reply docs/addon-dashboard.md]: ah yes, good point I will fix that

Comment (evilnick) [reply docs/addon-dashboard.md]: done

File: docs/addon-dashboard.md (hunk 2 above)
Comment (evilnick) [reply docs/addon-dashboard.md]: done

**Comment by evilnick** [comment]
@ktsakalozos I have made those changes if you want to give this one a +1

**Pull request merged**